hrms/hr/utils.py

```diff
@@ -345,35 +345,35 @@ def allocate_earned_leaves():
 
 	for e_leave_type in e_leave_types:
 		leave_allocations = get_leave_allocations(today, e_leave_type.name)
-
-		for allocation in leave_allocations:
-			if not allocation.leave_policy_assignment and not allocation.leave_policy:
-				continue
-
-			leave_policy = (
-				allocation.leave_policy
-				if allocation.leave_policy
-				else frappe.db.get_value(
-					"Leave Policy Assignment", allocation.leave_policy_assignment, ["leave_policy"]
+		if leave_allocations:
```

Comment:
Is this needed since get_leave_allocations will return an empty list IG?

If its needeed, A cleaner way to traverse would be this IMO

```suggestion
		if not leave_allocations:
			continue
```

Will save you some indentation for inner blocks. Also mentally you understand there is no operation in the current iteration.

Reply:
Added this since it printed an error when tested from the bench console.
Thanks for the suggestion! I've pushed the changes.